src/moq/transport.cpp

```diff
@@ -589,39 +567,39 @@ namespace moq {
            Span<const uint8_t> data) -> PublishTrackHandler::PublishObjectStatus {
            return SendObject(track_handler, priority, ttl, stream_header_needed, group_id, object_id, data);
        };
+       // Set the track handler for pub/sub
+
+       conn_it->second.pub_tracks_by_name[th.track_namespace_hash][th.track_name_hash] = std::move(track_handler);
    }
 
-   PublishTrackHandler::PublishObjectStatus Transport::SendObject(std::weak_ptr<PublishTrackHandler> track_handler,
+   PublishTrackHandler::PublishObjectStatus Transport::SendObject(const std::shared_ptr<PublishTrackHandler>& track_handler,
```

Comment:
It's 100% safe using weak as this is self. Why make it shared?

Reply:
Another recommendation of clang-tidy to use const ref weak_ptr, but that felt weird. The reality is that because we're using it so confidently as a weak ptr that we know will be valid, then we may as well pass the shared ptr since we are going to use it that way (so we save on `weak_ptr.lock()`), and since we don't store it we can pass it by const ref.

Reply:
If this function is always going to be called from a scope that ensures lifetime, what's wrong with a straight up const reference?